### PR TITLE
fix(ns): makes namespaces customizable

### PR DESCRIPTION
**Pull request by bartoszmajsak**
This PR allows customizing both prow namespace as well as the one where jobs run. 

Additionally, I improved both `config-gen.sh` and `update.sh` so that they can be invoked from anywhere, not only from within the directory they are in.

### CODE REVIEW

File: prow/nginx-ingress/ci/daemonset-tcp-udp-configMapNamespace-values.yaml [hunk 1]
```diff
@@ -3,12 +3,12 @@ controller:
   service:
```
Comment (dgn) [inline prow/nginx-ingress/ci/daemonset-tcp-udp-configMapNamespace-values.yaml:3]: these changes are not required. this directory is just a copy of the nginx chart, and this configmap in particular is never used by us

File: prow/nginx-ingress/ci/deployment-tcp-udp-configMapNamespace-values.yaml [hunk 2]
```diff
@@ -2,12 +2,12 @@ controller:
   service:
```
Comment (dgn) [inline prow/nginx-ingress/ci/deployment-tcp-udp-configMapNamespace-values.yaml:2]: same here, let's not change this, it's just going to get lost anyway when we update the nginx chart

**Review by dgn — APPROVED** [review]
lgtm, just two minor comments

File: prow/nginx-ingress/ci/daemonset-tcp-udp-configMapNamespace-values.yaml (hunk 1 above)
Comment (bartoszmajsak) [reply prow/nginx-ingress/ci/daemonset-tcp-udp-configMapNamespace-values.yaml]: Fixed 52b4eca

File: prow/nginx-ingress/ci/deployment-tcp-udp-configMapNamespace-values.yaml (hunk 2 above)
Comment (bartoszmajsak) [reply prow/nginx-ingress/ci/deployment-tcp-udp-configMapNamespace-values.yaml]: Fixed 52b4eca

**Comment by bartoszmajsak** [comment]
@dgn @jwendell I found some minor issues with the changes in the script, so hold on with marking it "ok-for-merge" :)

File: prow/create.sh [hunk 3]
```diff
@@ -2,22 +2,25 @@
 
 set -ex
 
+NAMESPACE=${NAMESPACE:-default}
+WORKER_NS=${WORKER_NS:-test-pods}
+
 # create test-pods namespace
-kubectl create namespace test-pods || echo Skipping
+kubectl create "${NAMESPACE}" test-pods || echo Skipping
```
Comment (jwendell) [inline prow/create.sh:9]: this is wrong

File: prow/gen-config.sh [hunk 4]
```diff
@@ -10,6 +15,7 @@ echo "#======================================
 #    Run gen-config.sh to regenerate.
 #======================================" > config.gen.yaml
 
-for file in config/*; do
-  cat "${file}" >> config.gen.yaml
+for file in "${DIR}"/config/*; do
+  # shellcheck disable=SC2016 ## in case of sed expression first '' is not an actual variable to be expanded
+  sed -e 's@${NAMESPACE}@'"${NAMESPACE}"'@' -e 's@${WORKER_NS}@'"${WORKER_NS}"'@' "${file}" >> "${DIR}"/config.gen.yaml
```
Comment (jwendell) [inline prow/gen-config.sh:20]: I'd put a `g` at the end of sed `s` command

File: prow/update.sh [hunk 5]
```diff
@@ -2,22 +2,28 @@
...
-kubectl -n default create configmap plugins --from-file=plugins.yaml=plugins.yaml --dry-run -o yaml | kubectl -n default replace configmap plugins -f -
+kubectl -n "${NAMESPACE}" create configmap config --from-file=config.yaml=config.gen.yaml --dry-run -o yaml | kubectl -n default replace configmap config -f -
+kubectl -n "${NAMESPACE}" create configmap plugins --from-file=plugins.yaml=plugins.yaml --dry-run -o yaml | kubectl -n default replace configmap plugins -f -
 
 # update deployments etc.
-for file in cluster/*; do
-  kubectl apply -f "${file}"
+for file in "${DIR}"/cluster/*; do
+  # shellcheck disable=SC2016 ## in case of sed expression first '' is not an actual variable to be expanded
+  sed -e 's@${NAMESPACE}@'"${NAMESPACE}"'@' 's@${WORKER_NS}@'"${WORKER_NS}"'@' "$file" | kubectl apply -f -
```
Comment (jwendell) [inline prow/update.sh:23]: same comment about adding  the `g` to sed

File: prow/create.sh (hunk 3 above)
Comment (bartoszmajsak) [reply prow/create.sh]: That's for the catch - that's what I meant in the comment above, but happy that someone has careful eyes :) Fixed in ef1fb7f910

File: prow/update.sh (hunk 5 above)
Comment (bartoszmajsak) [reply prow/update.sh]: Additionally, this line was missing an extra `-e` flag - added in 9cdd333

File: prow/gen-config.sh (hunk 4 above)
Comment (bartoszmajsak) [reply prow/gen-config.sh]: Good idea! Thank you. Added in 75ea8f3

**Review by jwendell — APPROVED** [review]

File: prow/cluster/hook_deployment.yaml [hunk 6]
```diff
@@ -41,6 +41,7 @@ spec:
         - --dry-run=false
         - --github-endpoint=https://api.github.com
         - --config-path=/etc/config/config.yaml
+        - --github-token-path=/etc/github/oauth
```
Comment (dgn) [inline prow/cluster/hook_deployment.yaml:44]: was this missing previously? is it required? our cluster is running without it

Comment (bartoszmajsak) [reply prow/cluster/hook_deployment.yaml:44]: It is of course running without - but you can see warnings in the hook logs about this soon becoming required. There are other warnings too.

Comment (jwendell) [reply prow/cluster/hook_deployment.yaml:44]: Are you sure the warnings are about this and not about not using ghproxy?

Comment (bartoszmajsak) [reply prow/cluster/hook_deployment.yaml:44]: About both

Comment (dgn) [reply prow/cluster/hook_deployment.yaml:44]: alright then, let's keep it in. yeah, the gh-proxy was on my list for a while, never got around to doing that

**Pull request merged**